lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2476,43 +2459,17 @@ def bar3d(self, x, y, z, dx, dy, dz, color=None,
 
         had_data = self.has_data()
 
-        if not cbook.iterable(x):
-            x = [x]
-        if not cbook.iterable(y):
-            y = [y]
-        if not cbook.iterable(z):
-            z = [z]
-
-        if not cbook.iterable(dx):
-            dx = [dx]
-        if not cbook.iterable(dy):
-            dy = [dy]
-        if not cbook.iterable(dz):
-            dz = [dz]
-
-        if len(dx) == 1:
-            dx = dx * len(x)
-        if len(dy) == 1:
-            dy = dy * len(y)
-        if len(dz) == 1:
-            dz = dz * len(z)
-
-        if len(x) != len(y) or len(x) != len(z):
-            warnings.warn('x, y, and z must be the same length.')
-
-        # FIXME: This is archaic and could be done much better.
-        minx, miny, minz = 1e20, 1e20, 1e20
-        maxx, maxy, maxz = -1e20, -1e20, -1e20
+        x, y, z, dx, dy, dz = np.broadcast_arrays(
+            np.atleast_1d(x), y, z, dx, dy, dz)
+        minx = np.min(x)
+        maxx = np.max(x + dx)
+        miny = np.min(y)
+        maxy = np.max(y + dy)
+        minz = np.min(z)
+        maxz = np.max(z + dz)
 
         polys = []
         for xi, yi, zi, dxi, dyi, dzi in zip(x, y, z, dx, dy, dz):
```

Comment:
Oh, it's `extend`, not `append` like I first thought, which is a bit more complicated.